numba/annotations/type_annotations.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import print_function, absolute_import
+import inspect
+import re
+from collections import Mapping, defaultdict
+import textwrap
+from contextlib import closing
+from numba.io_support import StringIO
+from numba import ir
+import numba.dispatcher
+import os
+import sys
+
+
+class SourceLines(Mapping):
+    def __init__(self, func):
+
+        try:
+            lines, startno = inspect.getsourcelines(func)
+        except IOError:
+            self.lines = ()
+            self.startno = 0
+        else:
+            self.lines = textwrap.dedent(''.join(lines)).splitlines()
```

Comment:
Technically, this code already existed before my changes (I moved the entire file to the new annotations subdir which is why the whole file looks new). My changes are almost all in the new html_annotate method, but we can talk about the rest of the file too :)